src/rust/ide/enso-protocol/src/file_manager.rs

```diff
@@ -0,0 +1,411 @@
+
+    /// Delete the specified watcher.
+    #[MethodInput=DeleteWatchInput,rpc_name="file/deleteWatch",result=delete_watch_result,
+    set_result=set_delete_watch_result]
+    fn delete_watch(&self, watch_id:Uuid) -> ();
+}
+}
+
+
+
+// =============
+// === Tests ===
+// =============
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use super::FileKind::RegularFile;
+
+    use json_rpc::messages::Message;
+    use json_rpc::messages::RequestMessage;
+    use json_rpc::test_util::transport::mock::MockTransport;
+    use serde_json::json;
+    use serde_json::Value;
+    use std::future::Future;
+    use utils::test::poll_future_output;
+    use utils::test::poll_stream_output;
+    use futures::task::LocalSpawnExt;
+
+    struct Fixture {
+        transport : MockTransport,
+        client    : Client,
+        executor  : futures::executor::LocalPool,
+    }
+
+    fn setup_fm() -> Fixture {
+        let transport = MockTransport::new();
+        let client    = Client::new(transport.clone());
+        let executor  = futures::executor::LocalPool::new();
+        executor.spawner().spawn_local(client.runner()).unwrap();
+        Fixture {transport,client,executor}
+    }
+
+    #[test]
+    fn test_notification() {
+        let mut fixture = setup_fm();
+        let mut events  = Box::pin(fixture.client.events());
+        assert!(poll_stream_output(&mut events).is_none());
+
+        let expected_notification = FilesystemEvent {
+            path : Path::new("./Main.luna"),
+            kind : FilesystemEventKind::Modified,
+        };
+        let notification_text = r#"{
+            "jsonrpc": "2.0",
+            "method": "filesystemEvent",
+            "params": {"path" : "./Main.luna", "kind" : "Modified"}
+        }"#;
+        fixture.transport.mock_peer_message_text(notification_text);
+        assert!(poll_stream_output(&mut events).is_none());
+
+        fixture.executor.run_until_stalled();
+
+        let event = poll_stream_output(&mut events);
+        if let Some(Event::Notification(n)) = event {
+            assert_eq!(n, Notification::FilesystemEvent(expected_notification));
+        } else {
+            panic!("expected notification event");
+        }
+    }
+
+    /// Tests making a request using file manager:
+    /// * creates FM client and uses `make_request` to make a request
+    /// * checks that request is made for `expected_method`
+    /// * checks that request input is `expected_input`
+    /// * mocks receiving a response from server with `result`
+    /// * checks that FM-returned Future yields `expected_output`
+    fn test_request<Fun, Fut, T>
+    ( make_request:Fun
+    , expected_method:&str
+    , expected_input:Value
+    , result:Value
+    , expected_output:T )
+    where Fun : FnOnce(&mut Client) -> Fut,
+          Fut : Future<Output = Result<T>>,
+          T   : Debug + PartialEq {
+        let mut fixture = setup_fm();
+        let mut fut     = Box::pin(make_request(&mut fixture.client));
```

Comment:
what is `fut`? Is it abbreviation from `future`? If so, it is like naming a variable `var`. Please use semantic naming.

Reply:
But we have such shortcuts in many places? Like all `off`s in AST (which were derived from scala AST)?

Reply:
@farmaazon I told in my comment that renaming it to `future` is a bad as naming a variable `var` ;) Use semantic, not syntactic naming!

Reply:
It's called `request` now :)